pkg/core/core.go

```diff
@@ -60,6 +61,11 @@ func SendDNSQuery(client *dns.Client, msg dns.Msg, dnsServerIP string) (*dns.Msg
 			logrus.Fatal("Please specify a DNS server IP explicitly with the `--dns-server-ip` flag.")
 		}
 		dnsServerIP = conf.Servers[0]
+
+	}
+	// If the server IP is IPv6, wrap it in square brackets to remove ambiguity from port number and address.
+	if net.ParseIP(dnsServerIP).To4() == nil {
+		dnsServerIP = "[" + dnsServerIP + "]"
```

Comment:
Recommendation: I would prefer to check if the parsed `dnsServerIP` includes a trailing port or not, if it contains a trailing port we should use the IP + the trailing port.

If it doesn't contain a trailing port we should append `:53` to the `dnsServerIP`.

And also, in another issue we should make it possible to pass a custom port to the `--dns-server-ip` flag.

Reply:
I've updated the PR to address your comment regarding the custom ports.
Besides that, I've cleaned up the implementation. `JoinHostPort(..)` does the exact same validation as I implemented before, but all in a single line 😄